File: qc-api/src/main/java/com/eager/questioncloud/exception/ErrorResponse.java
package com.eager.questioncloud.exception;

import lombok.Builder;
import lombok.Getter;
import org.springframework.http.ResponseEntity;

@Builder
@Getter
public class ErrorResponse {
    private int status;
    private String message;

    public static ResponseEntity<ErrorResponse> toResponse(CustomException e) {
        return ResponseEntity
            .status(e.getError().getHttpStatus())
            .body(ErrorResponse
                .builder()
                .status(e.getError().getHttpStatus())
                .message(e.getError().getMessage())
                .build());
    }
}

File: qc-api-container/src/main/kotlin/com/eager/questioncloud/exception/ErrorResponse.kt
package com.eager.questioncloud.exception

import com.eager.questioncloud.common.exception.CoreException
import org.springframework.http.ResponseEntity

class ErrorResponse(
    val status: Int,
    val message: String,
) {
    companion object {
        fun toResponse(e: CoreException): ResponseEntity<ErrorResponse> {
            return ResponseEntity
                .status(e.error.httpStatus)
                .body(ErrorResponse(e.error.httpStatus, e.error.message))
        }
    }
}